carmm/build/neb/bond_length_scan.py

```diff
@@ -99,9 +104,11 @@ def take_second(n):
 
         # apply bias in z-coordinate towards the surface atoms
         if z_bias:
-            # TODO: consult minimum distance from surface (can cause trouble for group)
             # min distance in Angstrom from surf atoms
-            z_threshold_min = surf_z + z_diff
+            if group_move:
```

Comment:
I'd reverse to a `if not group_move` and then you can remove the unnecessary pass.

Reply:
I removed this altogether, `z_threshold_min` applied before will affect all atoms in the `group_move` list